### DISCUSSION TITLE
organizations - automatic IDP redirect not working with existing realms

### DISCUSSION THREAD

**Discussion by yohan-hamon-pro** [keycloak/keycloak#34208] [Q&A] [answered]
Hi,
I am migrating from v24 to v25, and I am trying to implement the new feature Organizations on an existing realm (created before v25), but I cannot achieve to implement the automatic IDP redirection (according to email domain). I have modified the authentication flow according to the documentation here (https://www.keycloak.org/docs/latest/server_admin/#mapping-organization-claims_server_administration_guide) but it doesn't work.
The automatic redirection works fine with a new realm.
The only log I found is : `Flow 'Conditional Organization flow' failed`
Does anyone manage to make it work on existing realms ?
Regards,
Yohan

**Comment by yohan-hamon-pro** [comment, marked as answer]
I have found the issue. For the automatic IDP redirection to work we need the user to have no credentials configured. If credentials are configured the user is redirected to the classic authentication form... I think this is a bad behaviour when adding Organizations to an existing realm (where every users might already have credentials).

**Reply by brannon-bowden** [reply]
Thank you for this! Just spent many hours trying to understand what was happening. Were you able to figure out how to make it work on existing realms without deleting credentials?

**Reply by yohan-hamon-pro** [reply]
Hello,
No. The only solution for existing realm is to delete credentials. The authentication form continue to ask for password until you delete it.
